### PR TITLE
Add json format option for runner daemon logs

### LINKED ISSUE #157: Have the Cron Control Runner log in JSON
The Cron Control runner should jog in JSON to make it easier to parse the logs into Logstash for searchability.

### PR DESCRIPTION
**Pull request by hanifn**
# Description
Add JSON formatting option for logs in cron control runner daemon.
Fixes #157 

# Steps to Test or Reproduce

Make and run the runner with the `-log-format json` option flag. This will change log output to the format:
```
<timestamp> <namespace> <JSON>
Example:
2018-09-12T13:20:24.856Z cron-control:cron-control-runner {"level":"DEBUG","message":"Starting with 1 event-retreival worker(s) and 5 event worker(s)","type":"cron-control-runner"}
```

### CODE REVIEW

File: runner/logger.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		logger,
+	}
+}
+
+func (crl *CronRunnerLogger) Printf(format string, v ...interface{}) {
+	if crl.Format == "json" {
+		msgStr := fmt.Sprintf(format, v...)
+
+		// Should be using level specific functions to set the `Level` (i.e.: `Debugf()`) but we're
+		// doing it this way fit with how `logger` is used in `runner.go`
```
Comment (mjangda) [inline runner/logger.go]: Should we just change how `logger` is called? Or is it not worth it / too much effort?

File: runner/logger.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	if dest == "os.Stdout" {
+		if format == "json" {
+			logWriter = NewLogWriter("cron-control", "cron-control-runner", false)
+		} else {
+			// original log format
+			logWriter = NewLogWriter("cron-control", "cron-control-runner", true)
+		}
+		logger = log.New(logWriter, "", logOpts)
+	} else {
+		fileWriter := NewFileWriter(dest, "cron-control", "cron-control-runner")
```
Comment (mjangda) [inline runner/logger.go]: How come we don't have the format check here?

File: runner/logger.go (hunk 1 above)
Comment (hanifn) [reply runner/logger.go]: I think there's definitely value in making more "correct" but as it is now, there is only one way it is being used (as in all logs are basically `DEBUG`) so I wasn't sure if it was worth doing it now or waiting to till the use case expands a little. I don't really know the planned direction for this runner daemon

File: runner/logger.go (hunk 2 above)
Comment (hanifn) [reply runner/logger.go]: That's is a good point. I must have missed that. I'll add the format checks for that

**Comment by simonwheatley** [comment]
Naive question: Do we need to check with Systems that they are happy with the proposed JSON format?

**Comment by seanosh** [comment]
Tested and it produced output like:

```
2019-02-13T14:57:46.965Z cron-control:cron-control-runner {"level":"DEBUG","message":"Starting with 1 event-retreival worker(s) and 5 event worker(s)","type":"cron-control-runner"}
2019-02-13T14:57:46.965Z cron-control:cron-control-runner {"level":"DEBUG","message":"Retrieving events every 60 seconds","type":"cron-control-runner"}
2019-02-13T14:57:46.966Z cron-control:cron-control-runner {"level":"DEBUG","message":"started retriever 1\n","type":"cron-control-runner"}
2019-02-13T14:57:46.966Z cron-control:cron-control-runner {"level":"DEBUG","message":"started event worker 4\n","type":"cron-control-runner"}
2019-02-13T14:57:46.966Z cron-control:cron-control-runner {"level":"DEBUG","message":"started event worker 2\n","type":"cron-control-runner"}
2019-02-13T14:57:46.966Z cron-control:cron-control-runner {"level":"DEBUG","message":"started event worker 5\n","type":"cron-control-runner"}
2019-02-13T14:57:46.966Z cron-control:cron-control-runner {"level":"DEBUG","message":"started event worker 3\n","type":"cron-control-runner"}
2019-02-13T14:57:46.966Z cron-control:cron-control-runner {"level":"DEBUG","message":"started event worker 1\n","type":"cron-control-runner"}
2019-02-13T14:58:05.475Z cron-control:cron-control-runner {"level":"DEBUG","message":"\u003cheartbeat eventsSucceededSinceLast=0 eventsErroredSinceLast=0\u003e","type":"cron-control-runner"}
2019-02-13T14:58:35.475Z cron-control:cron-control-runner {"level":"DEBUG","message":"\u003cheartbeat eventsSucceededSinceLast=2 eventsErroredSinceLast=0\u003e","type":"cron-control-runner"}
2019-02-13T14:59:05.475Z cron-control:cron-control-runner {"level":"DEBUG","message":"\u003cheartbeat eventsSucceededSinceLast=0 eventsErroredSinceLast=0\u003e","type":"cron-control-runner"}
```

Not sure if the newlines/unicode `<` `>` are intentional/desired.

**Comment by darthhexx** [comment]
2 things:
1. We should get rid of the unnecessary characters.
2. This is going to break jobs stats, as gnaw processes the original output to garner them, so we'll need to coordinate the changes.

**Pull request closed without merging**